components/kyma-environment-broker/internal/orchestration/manager/upgrade_cluster.go

```diff
@@ -119,30 +120,36 @@ func (u *upgradeClusterFactory) CancelOperations(orchestrationID string) error {
 	return nil
 }
 
-// get current retrying operations, update state to pending and update other required params to storage
-func (u *upgradeClusterFactory) RetryOperations(orchestrationID string, schedule orchestration.ScheduleType, policy orchestration.MaintenancePolicy, updateMWindow bool) ([]orchestration.RuntimeOperation, error) {
+// get current retrying operations
+func (u *upgradeClusterFactory) RetryOperations(orchestrationID string) ([]orchestration.RuntimeOperation, error) {
 	result := []orchestration.RuntimeOperation{}
 	ops, _, _, err := u.operationStorage.ListUpgradeClusterOperationsByOrchestrationID(orchestrationID, dbmodel.OperationFilter{States: []string{orchestration.Retrying}})
 	if err != nil {
 		return nil, errors.Wrap(err, "while listing retrying operations")
 	}
 
 	for _, op := range ops {
-		if updateMWindow {
```

Comment:
Hi Laura, this `updateMWindow` is used to sync/update the maintenanceWindow right before the retrying operations are exactly retried. seems Hai's --now is an option, if --now is set to false, we still need to follow the maintenance window to retry the operations. Here is to consider in case the maintenance window got changed before the retry is triggered, so the latest maintenance window needs to be always synced right before the retry performed.